openquake/hazardlib/tests/gsim/check_gsim.py

```diff
@@ -54,12 +55,18 @@ def check_gsim(gsim_cls, datafile, max_discrep_percentage, debug=False):
         and a message pointing to a line with a test that failed will show up.
         If ``False`` the GSIM is executed in a vectorized way (if possible)
         and all the tests are executed even if there are errors.
+    :param instantiated:
+        If ``True`` the gsim_cls object is an instantiated GSIM class,
+        otherwise ``False``
 
```

Comment:
This is really ugly. Can we just pass a gsim instance, always? I would change `gsim_by_import_path` 
`return gsim_class` -> `return gsim_class()` and this would work for GSIMs without parameters. If you want this to work also with GSIMs with parameters you can try `return object.__new__(gsim_class)`.

Reply:
Sorry, I need more information here. How do I do this so as to avoid having to change every call to a GMPE in all of the other GMPE tests? If I do `return object.__new__(gsim_class)` then it passes instantiated objects (i.e. with the parameters) but then the uninstantiated objects (which is all the rest of them) don't work. Do I need to go and change them all to `NameEtAlYYYY()`? If not, and I have to place an if statement here (i.e. return `gsim_class()` if not with parameters, and `object.__new__(gsim_class)` with them) then I don't understand the advantages of that over the way this has been implemented in the current PR? Can you advise what I need to change and where?

Reply:
Let's do the following, which is a very minor change. Remove the argument `instantiated` from the `check_gsim` function and then replace the check 

```
￼     if instantiated:
￼         gsim = copy.deepcopy(gsim_cls)
￼     else:
￼         gsim = gsim_cls()
```
with
```
￼     if isinstance(gsim_cls, GroundShakingIntensityModel):
￼         gsim = copy.deepcopy(gsim_cls)
￼     else:
￼         gsim = gsim_cls()
```

in this way the user does not need to pass an argument than Python can figure out. I still do not like much having an argument `gsim_cls` which sometimes is a class and sometimes is an instance, but this can be fixed later on in another PR.

For the rest, LGTM. Please fix the conflict in the changelog and we will merge it.